climate_emotions_map/make_map.py

```diff
@@ -0,0 +1,277 @@
+    # do not show the hoverboxes here because for some reason they are not centered properly
+    fig.add_choropleth(
+        locations=df_to_plot[col_location],
+        geojson=survey_states,
+        z=df_to_plot[col_gradient],
+        zmin=vmin,
+        zmax=vmax,
+        colorscale=colormap,
+        colorbar_title=col_gradient.capitalize(),
+        name="main_map",
+        hoverinfo="none",  # no hoverbox but click events are still emitted (?)
+    )
+
+    # add outline for clicked state
+    if clicked_state is not None:
+        df_to_plot_clicked = df_to_plot[
+            df_to_plot[col_location] == clicked_state
+        ]
+        fig.add_choropleth(
+            locations=df_to_plot_clicked[col_location],
+            geojson=survey_states,
+            z=df_to_plot_clicked[col_gradient],
+            zmin=vmin,
+            zmax=vmax,
+            colorscale=colormap,
+            hoverinfo="skip",
+            name="clicked_state",
+            marker=clicked_state_marker,
+            showscale=False,
+        )
+
+    # add hover information
+    df_hover_data = state_abbrevs_long.merge(
+        df_to_plot,
+        on=col_location,
+    ).merge(
+        samplesizes_state,
+        on=col_location,
+    )
+    # if gradient only
+    customdata_cols = [col_location, "n"]
+    hovertemplate_extra = ""
+    # if gradient and scatter dots
+    if impact is not None and not show_impact_as_gradient:
+        customdata_cols.append(col_color_impact)
+        hovertemplate_extra = (
+            f"<br>{col_color_impact.capitalize()}: %{{customdata[2]:.2f}}"
+        )
+    fig.add_choropleth(
+        locations=df_hover_data["state_abbreviated"],
+        locationmode="USA-states",
+        customdata=df_hover_data[customdata_cols],
+        z=df_hover_data[col_gradient],
+        marker=dict(opacity=0),
+        name="hover_info",
+        hovertemplate=(
+            "<b>%{customdata[0]}</b>"
+            "<br>Sample size: %{customdata[1]}"
+            f"<br>{col_gradient.capitalize()}: %{{z:.2f}}"
+            f"{hovertemplate_extra}"
+            "<extra></extra>"
+        ),
+        showscale=False,
+    )
+
+    # add dots for impact data
+    if impact is not None and not show_impact_as_gradient:
+        vmin_impact = df_hover_data[col_color_impact].min()
+        vmax_impact = df_hover_data[col_color_impact].max()
+
+        # add dots one at a time to control color/size
+        for _, row in df_hover_data.iterrows():
+            fig.add_scattergeo(
+                locations=[row["state_abbreviated"]],
+                locationmode="USA-states",
+                marker={
+                    "size": row[col_color_impact] * impact_marker_size_scale,
+                    "color": px.colors.sample_colorscale(
+                        impact_colormap,
+                        (row[col_color_impact] - vmin_impact)
+                        / (vmax_impact - vmin_impact),
+                    ),
```

Comment:
this is very cool, but I feel like just relying on the size of the marker to encode the (relative) impact response is clearer to see. The colormap for impact is pretty close to the one for opinions (and we may not be able to find two maps that are completely orthogonal) - so they clash a little. And I don't think we (want to) have two separate color maps for opinion and impact either. 

So maybe just size? For extra happiness, you could consider using emojis instead of circles: https://community.plotly.com/t/i-want-to-use-custom-icon-for-the-scatter-markers-how-to-do-it/6644/6 (not sure if that would still be able to scale though, so maybe not)